code_getter/getter.py

```diff
@@ -375,6 +393,75 @@ def expand_line(self, s):
                     s = sublime.Region(s.begin(), line.end())
                     break
 
+        elif re.match(r"\s*#=", thiscmd) and not re.match(r".*=#\s*$", thiscmd):
+            indentation = re.match(r"^(\s*)", thiscmd).group(1)
+            endline = view.find(r"^" + indentation + r"=#", s.begin())
+            s = sublime.Region(s.begin(), view.line(endline.end()).end())
+        
+        elif re.match(r"\s*#>>", thiscmd) and not re.match(r".*#<<\s*$", thiscmd):
+            indentation = re.match(r"^(\s*)", thiscmd).group(1)
+            endline = view.find(r"^" + indentation + r"#<<", s.begin())
+            s = sublime.Region(s.begin(), view.line(endline.end()).end())
+
+        elif re.match(r".*].*$", thiscmd) and not re.match(r".*\[.*$", thiscmd):
+            s = self.backward_sq_bracket_expand(s, pattern=r"[\[](?=\s*)")
```

Comment:
It is not very accurate. Imagine we have a string `s = "]"`.

Reply:
Sry didn't realise it would be updating the pull as I make changes to my repository. The square bracket is a work in progress from today.

What is the correct way to do these pull requests? Creating a separate branch for each pull request?

Reply:
Yup. It is usually a better idea to create a new branch for pull requests.

Reply:
But at least in matlab there are no other uses for square brackets than array definitions so solving the issue you raised should be enough.

Reply:
I argue that we shouldn’t allow this though. The code is not readable with that style and you should first run a code formatter.

Reply:
that's actually how I format my matrices most of the time.

I personally want the result to be a live matlab interpreter where I can send commands as I write the file. But I will be happy to keep it in my fork with more brute force algorithms, leaving your elegant code as is.

![image](https://user-images.githubusercontent.com/61360439/100870034-3db9dc80-349e-11eb-9161-38d994899017.png)